### PR TITLE
Remove new relic

### PR DESCRIPTION
**Pull request by mathnogueira**
New Relic was not being used anymore.

This required some changes in the code, as some routines were strongly coupled with NewRelic. This PR:

1. Removes any reference to `WithSegment`
2. Removes NewRelic configuration and middleware
3. Fixes the problem with that test that keeps breaking our CI, however, it's a very bad solution and it will be addressed in the next PR.

### CODE REVIEW

File: api/clan.go [hunk 1]
```diff
@@ -368,92 +307,71 @@ func LeaveClanHandler(app *App) func(c echo.Context) error {
...
-					}
-					log.E(l, "Clan leave failed.", func(cm log.CM) {
-						cm.Write(zap.Error(err))
-					})
-				}
-				return err
-			}
-			return nil
-		})
+		tx, err = app.BeginTrans(c.StdContext(), l)
```
[the remark this answers is not in the document]
Comment (arthur29) [reply api/clan.go:310]: It will be done in the future

File: api/clan.go [hunk 2]
```diff
@@ -368,92 +307,71 @@ func LeaveClanHandler(app *App) func(c echo.Context) error {
...
 		if err != nil {
-			return FailWithError(err, c)
+			return FailWith(500, err.Error(), c)
```
[the remark this answers is not in the document]
Comment (mathnogueira) [reply api/clan.go:312]: I created the issue #76 and I will address it in a new PR.

File: api/clan.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (mathnogueira) [reply api/clan.go:310]: I created issue #76 and I will address it in a new PR.

File: api/clan_test.go [hunk 3]
```diff
@@ -1045,6 +1045,9 @@ var _ = Describe("Clan API Handler", func() {
 			)
 			Expect(err).NotTo(HaveOccurred())
 
+			// TODO: Replace this with something more reliable
```
[the remark this answers is not in the document]
Comment (mathnogueira) [reply api/clan_test.go:1048]: Issue #77

File: api/clan.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (mathnogueira) [reply api/clan.go:312]: This is the `echo.Context` and `c` is the convention that the community uses for this variable type

**Review by arthur29 — APPROVED** [review]

**Pull request merged**